new-log-viewer/src/App.tsx

```diff
@@ -10,11 +16,19 @@ import UrlContextProvider from "./contexts/UrlContextProvider";
  */
 const App = () => {
     return (
-        <UrlContextProvider>
-            <StateContextProvider>
-                <Layout/>
-            </StateContextProvider>
-        </UrlContextProvider>
+        <CssVarsProvider
```

Comment:
It's a bit tricky that we need to have NotificationContextProvider to be a direct / indirect child of CssVarsProvider so that JoyUI theming stuff can be correctly applied to the UI component like PopUpMessagesContainer. 

On the other hand, we need to have the NotificationContextProvider to wrap any children who need to publish / subscribe to the notifications, so we have to change the structure that way.

Are you suggesting that we can extract the UI stuff of the NotificationContextProvider as a separate component which subscribes from the provider and is to be placed inside Layout? Sounds like some enclaving territory to me lol. but I guess that works and makes the code cleaner. Let me know if i misunderstood

Reply:
Sorry i didn't see https://github.com/y-scope/yscope-log-viewer/pull/84#discussion_r1797206499 when I read this